scripts/kconfig: Introduce dt_node_ph_prop_path function

ydamigos opened the pull request:
Introduce dt_node_ph_prop_path function.
It  takes a node 'path' and a phandle property name and returns the path to the pointed-to node.

Needed by PR #73336

ydamigos committed:
scripts/kconfig: Introduce dt_node_ph_prop_path function

Introduce dt_node_ph_prop_path function.
It  takes a node 'path' and a phandle property name
and returns the path to the pointed-to node.

Signed-off-by: Ioannis Damigos <ioannis.damigos.uj@renesas.com>

ydamigos commented:
"Compliance Checks" failure seems unrelated to PR changes.

blauret commented:
> "Compliance Checks" failure seems unrelated to PR changes.

It's linked to the introdution of the othe PyLint version 3.2.x. Before that the lint passes.

ydamigos committed:
kconfigfunctions.py: Fix used-before-assignment error with pylint 3.2

CI reports error:
kconfigfunctions.py:143:11: E0601: Using variable 'edtlib' before
assignment (used-before-assignment)

Initialize edtlib to none when there is no edt.

Signed-off-by: Ioannis Damigos <ioannis.damigos.uj@renesas.com>
